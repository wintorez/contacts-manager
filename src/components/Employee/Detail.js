import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Header, List, Button, Divider } from 'semantic-ui-react';
import Page from '../Page';

const Detail = ({ location, history, loading, error, employee }) => (
  <Page
    title={employee ? `${employee.firstName} ${employee.lastName}` : undefined}
    location={location}
    loading={loading}
    error={error}
  >
    <Button floated="right" onClick={() => history.push('/employees')}>
      Back
    </Button>
    <Header as="h3" floated="left">
      {employee ? 'Employee Detail' : 'Employee Not Found'}
    </Header>
    <Divider clearing hidden />
    {employee && (
      <Fragment>
        <List relaxed="very">
          <List.Item>
            <List.Header>Name</List.Header>
            <List.Description>
              {`${employee.firstName} ${employee.lastName}`}
            </List.Description>
          </List.Item>
          <List.Item>
            <List.Header>Email</List.Header>
            <List.Description>
              <a href={`mailto:${employee.email}`}>{employee.email}</a>
            </List.Description>
          </List.Item>
          {employee.telephone && (
            <List.Item>
              <List.Header>Telephone</List.Header>
              <List.Description>
                <a href={`tel:${employee.telephone}`}>{employee.telephone}</a>
              </List.Description>
            </List.Item>
          )}
          {employee.department && (
            <List.Item>
              <List.Header>Department</List.Header>
              <List.Description>{employee.department}</List.Description>
            </List.Item>
          )}
          <List.Item>
            <Button
              onClick={() =>
                history.push(`/employees/${employee.employeeId}/edit`)
              }
            >
              Edit
            </Button>
          </List.Item>
        </List>
      </Fragment>
    )}
  </Page>
);

Detail.propTypes = {
  location: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  loading: PropTypes.bool,
  error: PropTypes.bool,
  employee: PropTypes.object
};

export default Detail;
