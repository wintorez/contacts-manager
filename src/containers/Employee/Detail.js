import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Detail from '../../components/Employee/Detail';

const mapStateToProps = (state, ownProps) => ({
  loading: state.status.loading,
  employee: state.employees.find(
    each => each.employeeId === ownProps.match.params.employeeId
  )
});

const mapDispatchToProps = (dispatch, ownProps) => ({});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Detail)
);
