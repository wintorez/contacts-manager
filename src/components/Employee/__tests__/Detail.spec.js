import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Detail from '../Detail';

const defaults = {
  history: {},
  employee: {
    employeeId: '1',
    firstName: 'Joe',
    lastName: 'Bloggs',
    email: 'jblogs@email.com',
    telephone: '+1-555-444-3333',
    department: 'Engineering'
  },
  location: { pathname: '/employees/1' }
};

describe('<Detail />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Detail {...defaults} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Detail {...defaults} loading />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with error', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Detail {...defaults} error />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
