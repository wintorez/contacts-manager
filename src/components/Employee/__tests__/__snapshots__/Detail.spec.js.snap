// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> renders correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  >
    <button
      className="ui right floated button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Back
    </button>
    <h3
      className="ui left floated header"
    >
      Employee Detail
    </h3>
    <div
      className="ui clearing hidden divider"
    />
    <div
      className="ui very relaxed list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Name
        </div>
        <div
          className="description"
        >
          Joe Bloggs
        </div>
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Email
        </div>
        <div
          className="description"
        >
          <a
            href="mailto:jblogs@email.com"
          >
            jblogs@email.com
          </a>
        </div>
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Telephone
        </div>
        <div
          className="description"
        >
          <a
            href="tel:+1-555-444-3333"
          >
            +1-555-444-3333
          </a>
        </div>
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Department
        </div>
        <div
          className="description"
        >
          Engineering
        </div>
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <button
          className="ui button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Edit
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Detail /> renders correctly when loading 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui active centered inline loader"
  />
</div>
`;

exports[`<Detail /> renders correctly with error 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui negative message"
  >
    Oops! Something went wrong! Please try again later.
  </div>
</div>
`;
